### PR TITLE
Add ability to use enterprise passenger

### PR DESCRIPTION
**Pull request by RLRickard**
Signed-off-by: RLRickard <ryanlrickard@gmail.com>

### Description

Allows consumers to utilize enterprise passenger

### Issues Resolved

NA

### Check List

- [ ] All tests pass. See <https://github.com/chef-cookbooks/community_cookbook_documentation/blob/master/TESTING.MD>
- [ ] New functionality includes testing.
- [ ] New functionality has been documented in the README if applicable
- [ ] All commits have been signed for the Developer Certificate of Origin. See <https://github.com/chef-cookbooks/community_cookbook_documentation/blob/master/CONTRIBUTING.MD>

### CODE REVIEW

**Comment by akrav** [comment]
@jtimberman @tas50 Can you guys review this PR?

File: recipes/source.rb [hunk 1]
```diff
@@ -43,9 +43,36 @@
   package %w(libssl-dev zlib1g-dev libapr1-dev libcurl4-gnutls-dev)
 end
 
-gem_package 'passenger' do
-  version node['passenger']['version']
-  gem_binary "#{node['passenger']['bin_dir']}/gem" unless node['passenger']['bin_dir'].nil?
+if node['passenger']['enterprise']
+  secrets_databag = chef_vauilt_item(node['passenger']['data_bag_name'], node['passenger']['data_bag_item'])
```
Comment (gravesb) [inline recipes/source.rb]: chef_vault_item.  this clearly needs to be tested better.

Comment (RLRickard) [inline recipes/source.rb]: I originally had these changes on a separate forked cookbook that I had tested against, but must have incorporated a few typos here and never tested against this branch.

I have fixed the typos and other missing pieces and have functionally tested my forked master branch.

**Comment by tas50** [comment]
@RLRickard If you want to rebase this I can merge it in.

**Comment by damacus** [comment]
Closing due to inactivity. 

If this is still an issue please reopen or open another issue. Alternatively drop by the #sous-chefs channel on the [Chef Community Slack](http://community-slack.chef.io/) and we'll be happy to help! 

Thanks,
Sous-Chefs

**Pull request closed without merging**